src/dune_rules/ordered_set_lang.ml

```diff
@@ -218,6 +218,8 @@ module Unexpanded = struct
 
   type t = ast generic
 
+  let of_include ~context sw = { ast = Ast.Include sw; loc = None; context }
```

Comment:
You can obtain the loc from `sw` here. This should simplify some of the manual loc book keeping.